grc/gui_qt/components/canvas/block.py

```diff
@@ -336,24 +357,39 @@ def paint(self, painter, option, widget):
 
                 display_value = ""
 
-                # Include the value defined by the user (after evaluation)
-                if not is_evaluated or self.show_param_val or not self.show_param_expr:
-                    display_value += item.options[value] if value in item.options else value  # TODO: pretty_print
+                show_value = (
+                    not is_evaluated or
+                    self.show_param_val or
+                    not self.show_param_expr
+                )
+                if show_value:
+                    display_value += (
+                        item.options[value]
+                        if value in item.options
+                        else value
+                    )
 
-                # Include the expression that was evaluated to get the value
                 if is_evaluated and self.show_param_expr:
-                    expr_string = value  # TODO: Truncate
+                    expr_string = value
 
-                    if display_value:  # We are already displaying the value
+                    if display_value:
                         display_value = expr_string + "=" + display_value
                     else:
                         display_value = expr_string
 
                 if len(display_value) > LONG_VALUE:
-                    display_value = display_value[:LONG_VALUE - 3] + '...'
+                    display_value = (
+                        display_value[: LONG_VALUE - 3] + "..."
+                    )
 
                 value_label = display_value
-                if (value is not None and item.hide == "none") or (item.dtype == 'id' and self.force_show_id) or self.core.is_dummy_block:
+                show_param = (
+                    (value is not None and item.hide == "none") or
+                    (item.dtype == "id" and self.force_show_id) or
+                    self.core.is_dummy_block or
+                    (item.dtype == "id" and self.core.key == "options")
```

Comment:
This is one of two changes in this PR. The rest is all whitespace, comments, formatting, ....

Reply:
seems reasonable, but I'm moving the common parts of the two checks into a small method, because this is the kind of thing one forgets to change when making change at one end.